src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -334,14 +335,24 @@ public override bool IsBindingRequired
         {
             get
             {
-                if (MetadataKind == ModelMetadataKind.Property)
-                {
-                    return BindingMetadata.IsBindingRequired;
-                }
-                else
+                if (!_isBindingRequired.HasValue)
                 {
-                    return false;
+                    if (MetadataKind == ModelMetadataKind.Type)
+                    {
+                        _isBindingRequired = false;
```

Comment:
Separately, `IsBindingAllowed` has the same problem with parameter metadata as `IsBindingRequired`. Should probably be fixed while we're looking at this.